### PR TITLE
Proxy Server Sent Events

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

File: __tests__/proxy-sse-test.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+          mockOnMessage();
+          if (callCount === 2) {
+            resolve();
+          }
+        }
+      });
+      expect(mockOnMessage).toHaveBeenCalledTimes(2);
+    });
+
+    it("forwards data with a similar interval", async () => {
```
Comment (juliangieseke) [inline __tests__/proxy-sse-test.ts]: 🗣  …in same order…?

Comment (juliangieseke) [reply __tests__/proxy-sse-test.ts]: otherwise I'd check for a range between 700-900, not >200? 🤔

File: __tests__/proxy-sse-test.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        }
+      });
+
+      expect(mockOnMessage).toHaveBeenCalledTimes(2);
+      const callTimeFirst = mockOnMessage.mock.calls[0][0];
+      const callTimeSecond = mockOnMessage.mock.calls[1][0];
+      expect(callTimeSecond - callTimeFirst).toBeGreaterThanOrEqual(200);
+    });
+
+    it("forwards typed data events", async () => {
```
Comment (juliangieseke) [inline __tests__/proxy-sse-test.ts]: Even if the tests before are implicitly testing untyped messages, can you add a test explicitly testing untyped? 🤔

File: __tests__/proxy-sse-test.ts (hunk 1 above)
Comment (DanielMSchmidt) [reply __tests__/proxy-sse-test.ts]: I don't check for the order, need to do another test then

**Comment by DanielMSchmidt** [comment]
@juliangieseke I addressed your change requests, could you review again?

**Comment by juliangieseke** [comment]
'flicts :(

**Comment by DanielMSchmidt** [comment]
@juliangieseke Could you please review this?

**Comment by nLight** [comment]
Flicts!

File: __tests__/proxy-sse-test.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
+jest.unmock("eventsource");
+
+import EventSource from "eventsource";
+import createServer from "./helpers/createServer";
+import MockServerController from "./helpers/mockServerController";
+
+interface IEventListener {
+  cb: (evt: Event) => void;
```
Comment (Fabs) [inline __tests__/proxy-sse-test.ts]: Where does the type Event come from? EventSource?

File: __tests__/proxy-sse-test.ts [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+import EventSource from "eventsource";
+import createServer from "./helpers/createServer";
+import MockServerController from "./helpers/mockServerController";
+
+interface IEventListener {
+  cb: (evt: MessageEvent) => void;
+  type: string;
+}
+interface ICallbacks {
+  onmessage?: (evt: Event, resolve: (...any) => void) => void;
```
Comment (Fabs) [inline __tests__/proxy-sse-test.ts]: Here it should also be MessageEvent right?

**Comment by DanielMSchmidt** [comment]
@Fabs @juliangieseke could you review again? 😇

**Pull request merged**